File: src/Entity/Cour.php
<?php

namespace App\Entity;

use App\Repository\CourRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=CourRepository::class)
 */
class Cour
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /** @Assert\Length(
     *     min=3,
     *     minMessage = "Le nom du cour doit au minimum {{ limit }} characters long",
     *
     *     )
     * @ORM\Column(type="string", length=50)
     */
    private $nomCour;

    /** @Assert\Length(
     *     min=3,
     *     minMessage = "Le nom de l Enseignant doit au minimum {{ limit }} characters long",
     *
     *     )
     * @ORM\Column(type="string", length=50)
     */
    private $nomEnseignant;

    /** @Assert\Length(
     *     min=3,
     *     minMessage = "La description doit au minimum {{ limit }} characters long",
     *
     *     )
     * @ORM\Column(type="string", length=50)
     */
    private $description;

    /**
     * @ORM\Column(type="string", length=50, nullable=true)
     */
    private $img;

    /**
     * @ORM\Column(type="integer", nullable=true)
     */
    private $idD;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomCour(): ?string
    {
        return $this->nomCour;
    }

    public function setNomCour(string $nomCour): self
    {
        $this->nomCour = $nomCour;

        return $this;
    }

    public function getNomEnseignant(): ?string
    {
        return $this->nomEnseignant;
    }

    public function setNomEnseignant(string $nomEnseignant): self
    {
        $this->nomEnseignant = $nomEnseignant;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getImg(): ?string
    {
        return $this->img;
    }

    public function setImg(string $img): self
    {
        $this->img = $img;

        return $this;
    }

    public function getIdD(): ?int
    {
        return $this->idD;
    }

    public function setIdD(?int $idD): self
    {
        $this->idD = $idD;

        return $this;
    }
}

File: src/com/mycompany/myapp/entities/Cour.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author 21655
 */
public class Cour {

      private int id ;
    private String nomCour;
    private String nomEnseignant;
    private String description;
    private String img;
    private int idD;

    public Cour(int id, String nomCour, String nomEnseignant, String description, String img, int idD) {
        this.id = id;
        this.nomCour = nomCour;
        this.nomEnseignant = nomEnseignant;
        this.description = description;
        this.img = img;
        this.idD = idD;
    }

    public Cour() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNomCour() {
        return nomCour;
    }

    public void setNomCour(String nomCour) {
        this.nomCour = nomCour;
    }

    public String getNomEnseignant() {
        return nomEnseignant;
    }

    public void setNomEnseignant(String nomEnseignant) {
        this.nomEnseignant = nomEnseignant;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    public int getIdD() {
        return idD;
    }

    public void setIdD(int idD) {
        this.idD = idD;
    }

    @Override
    public String toString() {
        return "Cour{" + "id=" + id + ", nomCour=" + nomCour + ", nomEnseignant=" + nomEnseignant + ", description=" + description + ", img=" + img + ", idD=" + idD + '}';
    }

    

    
    
    
}
